File: chapter02/02.03.1/solution/HousingData.js
const seaborn = require('seaborn');
const pandas = require('pandas');
// Describe: Function to create a univariate and bivariate analysis on dataset.
function performAnalysis(datasetPath) {
  // Load the dataset using pandas
  const dataset = pandas.read_csv(datasetPath);

  // Perform univariate analysis
  seaborn.displot(dataset);

  // Perform bivariate analysis
  //CRIM,ZN
  seaborn.jointplot(x='column1', y='column2', data=dataset);
}

performAnalysis('/Users/mickknutson/Documents/workspace/BASELOGIC/github_copilot_master_class/chapter02/02.03.1/solution/HousingData.csv');

File: chapter02/02.03.1/solution/HousingData.py
# Describe: Function to create a univariate and bivariate analysis on dataset.
import pandas as pd
import seaborn as sns

def analyze_data():
    # Load the dataset
    data = pd.read_csv('/Users/mickknutson/Documents/workspace/BASELOGIC/github_copilot_master_class/chapter02/02.03.1/solution/HousingData.csv')

    # Univariate analysis
    # Example: Histogram of a numerical variable
    #sns.histplot(data['SalePrice'])
    sns.displot(data);

    # Bivariate analysis
    # Example: Scatter plot of two numerical variables
    #sns.scatterplot(x='GrLivArea', y='SalePrice', data=data)

    # Show the plots
    #sns.plt.show()

# Call the function to perform the analysis
analyze_data()
